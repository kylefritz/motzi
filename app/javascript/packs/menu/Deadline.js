import React from "react";
import moment from "moment";

export default function Deadline({ menu }) {
  const { deadlineDay } = menu;

  const now = moment();
  const deadline = moment(menu.deadline);
  const pastDeadline = now > deadline;
  if (!pastDeadline) {
    return null;
  }

  return (
    <div className="alert alert-warning text-center" role="alert">
      <h4 className="alert-heading">Ordering is closed for this menu</h4>
      Order by {deadlineDay} at midnight for each week's menu.
    </div>
  );
}
