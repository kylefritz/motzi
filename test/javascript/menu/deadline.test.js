import React from "react";
require("../configure_enzyme");
import { shallow } from "enzyme";

import Deadline from "menu/Deadline";
import orderData from "./order-data";
import moment from "moment";

test("Before deadline snapshot", () => {
  const wrapper = shallow(<Deadline {...orderData} />);
  expect(wrapper).toMatchSnapshot();
});

test("After deadline snapshot", () => {
  orderData.menu.deadline = moment().add(-1, "days");
  const wrapper = shallow(<Deadline {...orderData} />);
  expect(wrapper).toMatchSnapshot();
});
