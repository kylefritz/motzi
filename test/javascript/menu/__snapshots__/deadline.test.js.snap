// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`After deadline snapshot 1`] = `
<div
  className="alert alert-warning text-center"
  role="alert"
>
  <h4
    className="alert-heading"
  >
    Ordering is closed for this menu
  </h4>
  Order by 
  Sunday
   at midnight for each week's menu.
</div>
`;

exports[`Before deadline snapshot 1`] = `""`;
